spack/package.py

```diff
@@ -24,18 +24,41 @@ class Oommf(Package):
     contributors to OOMMF are Mike Donahue, and Don Porter.
 
     Summary taken from OOMMF documentation https://math.nist.gov/oommf/
+
+    OOMMF home page: "https://math.nist.gov/oommf/"
+
+    OOMMF as a git repository: https://github.com/fangohr/oommf
+
+    Versions ending with "-vanilla" indicate that the sources are taken
+    directly from https://math.nist.gov/oommf/dist/ . All other versions are
+    from the https://github.com/fangohr/oommf (which includes the "-vanilla"
+    sources, and adds additional OOMMF extensions). See
+    https://github.com/fangohr/oommf for details.
     """
 
     homepage = "https://math.nist.gov/oommf/"
-    url = "https://math.nist.gov/oommf/dist/oommf20a2_20200608-hotfix.tar.gz"
+    url = "https://github.com/fangohr/oommf/archive/refs/tags/20a1_20180930_ext.tar.gz"
 
     maintainers = ["fangohr"]
 
+    version('20a1_20180930_ext',
+            sha256='18bf9bd713c7ee6ced6d561ce742d17e0588ae24ef2e56647a5c8a7853e07a4c',
+            preferred=True)
+
     version(
-        "20200608-hotfix",
+        "20200608-hotfix-vanilla",
         sha256="5c349de6e698b0c2c5390aa0598ea3052169438cdcc7e298068bc03abb9761c8",
+        url = "https://math.nist.gov/oommf/dist/oommf20a2_20200608-hotfix.tar.gz"
     )
 
+    # Deprecated versions have never been tested with spack.
+    version('20a1_20180930', deprecated=True,
+            sha256='c871e0dbb1522c3c1314af6c084b90cdbe69fd869b55ac94443851b74f818ed2')
+    version('20170929a0', deprecated=True,
+            sha256='3439d1c9e95cc7395bc2e2330bba8cf198585d1b350251ea8561c1554ff8c7fd')
+    version('20160930b1', deprecated=True,
+            sha256='363006f549bb63a39564fafc18b52342a14c1c3769c214467a39f72a0c0be36b')
```

Comment:
Where do these releases come from? Looks like older releases on GitHub but the names are slightly different (missing the first part + underscore). Does this have any effect?

Reply:
Hm, good point. I had assumed spack would retrieve the releases we have (https://github.com/fangohr/oommf/releases), but maybe it went for the tags (https://github.com/fangohr/oommf/tags).

I think given the version name (first argument), spack can download the right file. The question we should discuss though is for which files/releases do we want to make this possible? I guess the three versions on https://github.com/fangohr/oommf/releases would be good (plus the new 20a2 when it is finished) ?

Reply:
I had another look, and have tidied up the code a little bit, and also made the version names more uniform.

I included some older versions as deprecated - we should ignore those; they are only there because it is easy to add them know - maybe it will help somebody at some point, but testing all of them would be substantial effort now with potentially little gain.

The output of `spack info oommf` now contains:
<pre>
Preferred version:  
    20a1_20180930_ext               https://github.com/fangohr/oommf/archive/refs/tags/20a1_20180930_ext.tar.gz

Safe versions:  
    20a2_20200608-hotfix-vanilla    https://math.nist.gov/oommf/dist/oommf20a2_20200608-hotfix.tar.gz
    20a1_20180930_ext               https://github.com/fangohr/oommf/archive/refs/tags/20a1_20180930_ext.tar.gz
</pre>

Reply:
Discussing which versions to include will be easier in person.